### ISSUE TITLE
Issue registering singleton

### ISSUE THREAD

**Issue by heathbar** [Microsoft/tsyringe#27]
I believe there is an issue with the `container.registerSingleton` method.

Suppose I have a class that depends on a singleton service as follows
```TypeScript
@singleton()
class MyService { ... }

class MyClass {
  constructor(private myService: MyService) { }
}
```
Then suppose I wish to mock this service in a test.
```TypeScript
container.registerSingleton(MyService, MyServiceMock);
const myClass = container.resolve(MyClass);
```
This fails as `MyClass` is instantiated with `MyService` rather than `MyServiceMock`. I believe the issue is at the bottom of the `registerSingleton` method shown here for reference:
https://github.com/Microsoft/tsyringe/blob/a1df21283d4e19ef7006848a800a6e9caa524c9b/src/dependency-container.ts#L70-L88

I believe lines 85-87 are the issue. `useClass: from` should be `useClass: to`. Or am I misusing this method?

**Comment by Xapphire13** [comment]
Nice catch. I think your reasoning is sound. When I get a chance I will add your repro code to our tests and give it a look.

**Comment by heathbar** [comment]
In the mean time, this seems to be a valid workaround:
```TypeScript
container.registerInstance(MyService, container.resolve(MyServiceMock));
```
Where `MyServiceMock` is registered as a `@singleton()`

**Comment by Xapphire13** [comment]
So I looked more into it, and as you can see here:
https://github.com/Microsoft/tsyringe/blob/a1df21283d4e19ef7006848a800a6e9caa524c9b/src/dependency-container.ts#L69

The original intent wasn't to support class redirects. However, after seeing your use-case, I think it's a really good idea and will push a PR for it shortly

**Comment by Xapphire13** [comment]
This is fixed in `v3.2.0`
